File: product-service/src/main/java/com/intexsoft/webshop/productservice/model/Image.java
package com.intexsoft.webshop.productservice.model;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Entity
@Table(name = "image")
@Data
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Image {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;
    @Column(name = "filePath", nullable = false, length = 512, unique = true)
    String filePath;
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    @ManyToOne(fetch = FetchType.LAZY)
    Product product;
}

File: product-service-kt/src/main/kotlin/com/intexsoft/webshop/productservicekt/model/Image.kt
package com.intexsoft.webshop.productservicekt.model;

import jakarta.persistence.*;

@Entity
@Table(name = "image")
class Image(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0,
    @Column(name = "filePath", nullable = false, length = 512, unique = true)
    var filePath: String,
    @ManyToOne(fetch = FetchType.LAZY)
    var product: Product?
)